lib/grape/middleware/base.rb

```diff
@@ -21,9 +21,17 @@ def call(env)
 
       def call!(env)
         @env = env
+        error_class = nil
         before
-        @app_response = @app.call(@env)
-        after || @app_response
+        begin
+          @app_response = @app.call(@env)
+        rescue => e
+          error_class = e
+        ensure
+          after_response = after
+        end
+
+        after_response || (error_class ? (fail error_class) : @app_response)
       end
```

Comment:
Hmmm... wait. Looks like the old behavior *was* to use the response. So let me take this back, we should preserve the response if it was there. I guess it should be closer to this:

```ruby
begin
  @app_response = @app.call(@env)
ensure
  @app_response = after || @app_response
end
```